engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -962,8 +963,11 @@ private void redoPartitions(PMapping currentMapping, String note)
       mainPC.operatorIdToPartition.put(p.getId(), newPartition);
     }
 
+    // Get the deepest down stream mapping
+    PMapping downStreamMapping = currentMapping;
```

Comment:
If partitionContexts linked hashmap exists then downStreamMapping be the last element of hasmap otherwise downStreamMapping be the currentMapping.
Lets say, the physical dag before repartition as below :
A1 -> B1 -> C1 -> D1 and parallel partition between A -> B and B -> C.

If A is repartitioned to 3 then 
downStreamMapping would be the Operator C's mapping.